src/Native/System.Security.Cryptography.Native/pal_asn1.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_asn1.h"
+
+#include <openssl/objects.h>
+
+static_assert(NID_undef == 0, "Managed code relies on NID_undef == 0");
+static_assert(NID_secp224r1 == 713, "Managed code relies on NID_secp224r1 == 713");
+static_assert(NID_secp384r1 == 715, "Managed code relies on NID_secp384r1 == 715");
+static_assert(NID_secp521r1 == 716, "Managed code relies on NID_secp521r1 == 716");
+
+extern "C" int64_t Asn1IntegerGet(const ASN1_INTEGER* a)
+{
+    return ASN1_INTEGER_get(a);
+}
+
+extern "C" const ASN1_OBJECT* ObjTxt2Obj(const char* s)
+{
+    return OBJ_txt2obj(s, true);
+}
+
+extern "C" int32_t ObjObj2Txt(char* buf, int32_t buf_len, const ASN1_OBJECT* a)
+{
+    return OBJ_obj2txt(buf, buf_len, a, true);
+}
+
+extern "C" const ASN1_OBJECT* GetFriendlyNameSharedObject(const char* friendlyName)
```

Comment:
`GetObjectDefinitionByName`, maybe?  This name feels like it's getting the friendly name out, vs using the friendly name as the input.